src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/SchemaUtils.java

```diff
@@ -0,0 +1,539 @@
+      stringBuilder.append(" WHERE");
+
+      for (String tableName : spannerTableNames) {
+        stringBuilder.append(" TABLE_NAME=\"");
+        stringBuilder.append(tableName);
+        stringBuilder.append("\"");
+        stringBuilder.append(" OR");
+      }
+
+      // Remove the last " OR".
+      stringBuilder.setLength(stringBuilder.length() - 3);
+    }
+  }
+
+  private static Map<String, SpannerTable> getSpannerTableByName(
+    DatabaseClient databaseClient, Set<String> spannerTableNames,
+    Map<String, Set<String>> spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName) {
+
+    StringBuilder queryInformationSchemaColumnsSqlStringBuilder = new StringBuilder(
+      "SELECT TABLE_NAME, COLUMN_NAME, ORDINAL_POSITION, SPANNER_TYPE "
+        + "FROM INFORMATION_SCHEMA.COLUMNS");
+
+    // Skip the columns of the tables that are not tracked by Change Streams.
+    appendWhereTableNamesEqualToSql(queryInformationSchemaColumnsSqlStringBuilder,
+      spannerTableNames);
+
+    final ResultSet columnsResultSet =
+      databaseClient
+        .singleUse()
+        .executeQuery(
+          Statement.newBuilder(queryInformationSchemaColumnsSqlStringBuilder.toString())
+            .build());
+    Map<String, List<SpannerColumn>> spannerColumnsByTableName = new HashMap<>();
+    while (columnsResultSet.next()) {
+      final String tableName = columnsResultSet.getString(INFORMATION_SCHEMA_TABLE_NAME);
+      final String columnName = columnsResultSet.getString(INFORMATION_SCHEMA_COLUMN_NAME);
+      // Skip if the columns of the table is tracked explicitly, and the specified column is not
+      // tracked.
+      if (spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName.containsKey(tableName)
+        && !spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName
+        .get(tableName).contains(columnName)) {
+        continue;
+      }
+
+      final int ordinalPosition = (int) columnsResultSet.getLong(INFORMATION_SCHEMA_ORDINAL_POSITION);
+      final String spannerType = columnsResultSet.getString(INFORMATION_SCHEMA_SPANNER_TYPE);
+      spannerColumnsByTableName.putIfAbsent(tableName, new ArrayList<>());
+      final SpannerColumn spannerColumn = SpannerColumn.create(columnName,
+        informationSchemaTypeToSpannerType(spannerType), ordinalPosition);
+      spannerColumnsByTableName.get(tableName).add(spannerColumn);
+    }
+
+    StringBuilder queryInformationSchemaKeyColumnsSqlStringBuilder = new StringBuilder(
+      "SELECT TABLE_NAME, COLUMN_NAME, CONSTRAINT_NAME FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE");
+    // Skip the columns of the tables that are not tracked by Change Streams.
+    appendWhereTableNamesEqualToSql(queryInformationSchemaKeyColumnsSqlStringBuilder,
+      spannerTableNames);
+
+    final ResultSet keyColumnsResultSet =
+      databaseClient
+        .singleUse()
+        .executeQuery(
+          Statement.newBuilder(queryInformationSchemaKeyColumnsSqlStringBuilder.toString())
```

Comment:
You can use `Statement.of(...)` if there are no parameters

Reply:
Done.